edgexfoundry/edgex-go
# Add dockerfile to build snap

This provides a simple Dockerfile (and script which builds the docker image) that is to be used by CI to build the snap. This removes the need to build the snap on Ubuntu 16.04, and the build for the snap may be run on any OS that runs docker.

To test, clone this branch and just run:

```bash
snap/build.sh
```

After a while you should see output similar to this:

```
[[ lots of other log output ]]
Staging mongo-config 
Staging mongodb 
Priming config-common 
Priming go 
Priming consul 
Priming curl 
Priming glide 
Priming edgex-go 
Priming mongo-config 
Priming mongodb 
Determining the version from the project repo (version-script).
The version has been set to '0.5.2-20180917+11f7d4b'
Snapping 'edgexfoundry' ...

Snapped edgexfoundry_0.5.2-20180917+11f7d4b_amd64.snap
```

The build is currently entirely done inside the container, and whether or not to do the release is determined by a few Jenkins specific environment variables, i.e. `JENKINS_URL` and `JOB_NAME`. If `JENKINS_URL` exists, then we copy the snap store login file into the container. Then we check if the JOB_NAME matches a stage job, and if it does then set a environment variable so the entrypoint script will attempt to release the snap on the specified snap channel.

See the associated ci-management job: https://github.com/edgexfoundry/ci-management/pull/211

This fixes #557
